iotdb-0.13/src/main/java/cn/edu/tsinghua/iot/benchmark/iotdb013/IoTDBSession.java

```diff
@@ -137,30 +137,35 @@ public Status insertOneBatchByRecords(Batch batch) {
         batch.getDeviceSchema().getSensors().stream()
             .map(Sensor::getName)
             .collect(Collectors.toList());
-
-    for (Record record : batch.getRecords()) {
-      deviceIds.add(deviceId);
-      times.add(record.getTimestamp());
-      measurementsList.add(sensors);
-      valuesList.add(record.getRecordDataValue());
-      typesList.add(
-          constructDataTypes(
-              batch.getDeviceSchema().getSensors(), record.getRecordDataValue().size()));
+    while (true) {
+      for (Record record : batch.getRecords()) {
+        deviceIds.add(getDevicePath(batch.getDeviceSchema()));
+        times.add(record.getTimestamp());
+        measurementsList.add(sensors);
+        valuesList.add(record.getRecordDataValue());
+        typesList.add(
+            constructDataTypes(
+                batch.getDeviceSchema().getSensors(), record.getRecordDataValue().size()));
+      }
+      if (!batch.hasNext()) {
+        break;
+      }
+      batch.next();
```

Comment:
for(;batch.hasnext();batch.next()) ?

Reply:
For old version `Batch`, batch.hasnext() always return false, because of the index is initially pointed to 0, not -1.
Therefore, the first batch.hasNext() check must occur after the first processing.
(By the way, if initially set index to -1, then each references of `Batch` in history code needs to be edit)